docs/migrate-from-sidecar-to-ambient/migration.adoc

```diff
@@ -0,0 +1,1604 @@
+
+**Option 2: Ztunnel + Waypoint Proxies**
+
+- **Full feature set**: All Istio traffic management and security features
+- **L7 policies**: HTTP-level authorization, routing, telemetry
+- **Gateway API support**: Modern traffic management APIs
+- **Performance cost**: Similar to sidecar mode for L7 processing
+
+**When Waypoints Are Used vs Not Used:**
+
+**Automatic Waypoint Activation occurs when:**
+1. **Namespace labeling**: Adding `istio.io/use-waypoint=<waypoint-name>` label to a namespace
+2. **Service labeling**: Adding the label to specific services for granular control
+3. **Pod labeling**: Adding the label to individual pods (requires `istio.io/waypoint-for: workload`)
+
+**Traffic Flow Decision Tree:**
+- **Without waypoint labels**: Traffic flows directly through ztunnel (L4-only processing)
+- **With waypoint labels**: Traffic is intercepted and routed through waypoint proxies for L7 processing
+- **Mixed environments**: Some services use waypoints while others remain ztunnel-only
+
+**Decision Criteria:**
+
+- **Use waypoints if**: You need VirtualServices, HTTPRoutes, L7 AuthorizationPolicies, fault injection, or HTTP-level observability
+- **Skip waypoints if**: You only need mTLS, L4 AuthorizationPolicies, and basic connectivity
+- **Hybrid approach**: Deploy waypoints only for services requiring L7 features
+
+**Important**: Waypoint activation is explicit, deploying a waypoint Gateway resource alone does not affect traffic flow. You must label namespaces, services, or pods to direct traffic through waypoints.
+
+For detailed waypoint configuration guidance, see link:../common/istio-ambient-waypoint.adoc[Waypoint Proxy Documentation].
+
+[[compatibility-validation]]
+=== 2.2 Compatibility between Sidecar and Ambient Modes
+
+**Supported Features:**
+
+- All L4 traffic policies (AuthorizationPolicy, PeerAuthentication)
+- L7 traffic management (VirtualService, DestinationRule converted to HTTPRoute)
+- Security policies (RequestAuthentication, AuthorizationPolicy)
+- Telemetry and observability features
+
+**Unsupported Features** (migration blockers):
+
+- Multi-cluster mesh configurations
+- Virtual Machine (VM) workload integration
+- SPIRE integration for identity federation
+
+For additional compatibility considerations refer to:
+- link:../common/coexistence.md [Sidecar-Ambient Coexistence Guidelines]
+
+[[pre-migration-checklist]]
+=== 2.3 Pre-Migration Checklist
+
+**Environment Validation:**
+[source,console]
+----
+# Verify Sail Operator is deployed
+kubectl get pods -n sail-operator
+----
+
+[source,console]
+----
+# Check Istio control plane version
+kubectl get istio -n istio-system
+NAME      NAMESPACE      PROFILE   REVISIONS   READY   IN USE   ACTIVE REVISION   STATUS    VERSION   AGE
+default   istio-system             1           1       1        default           Healthy   v1.27.0   20m
+----
+
+[source,console]
+----
+# Verify no unsupported features are in use
```

Comment:
In the next sections, the `virtualservice` is used. So what does this step mean?
If I verify that I am using unsupported features (`virtualservice` cr) can I continue or not? 

In my POV, this step can causes confusion what next if I am using virtualservice in my app project.